include/tvm/runtime/data_type.h

```diff
@@ -237,6 +241,49 @@ class DataType {
     }
   }
 
+  /*!
+   * \brief Return the widest type from a given array.
+   *
+   * Given an array of types, return the one of those types that can
+   * hold all values representable in the other types, or "Void" if
+   * such type is not present in the array.
+   */
+  static DataType WidestOf(const std::vector<DataType>& types) {
```

Comment:
is it possible to use template trick or overloading to handle two/three case, mainly avoid the vector passing

Reply:
Would `std::initializer_list` be ok?